instrumentation/messaging/src/main/java/brave/messaging/MessagingRequest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.messaging;
+
+import brave.internal.Nullable;
+
+/**
+ * Abstract request type used for parsing and sampling of messaging producers and consumers.
+ *
+ * @see ProducerRequest
+ * @see ConsumerRequest
+ * @since 5.9
+ */
+public abstract class MessagingRequest {
+  /**
+   * Returns the underlying messaging request object. Ex. {@code javax.jms.Message}
+   *
+   * <p>Note: Some implementations are composed of multiple types, such as a message and a
+   * destination. Moreover, an implementation may change the type returned due to refactoring.
+   * Unless you control the implementation, cast carefully (ex using {@code instance of}) instead of
+   * presuming a specific type will always be returned.
+   *
+   * @since 5.9
+   */
+  public abstract Object unwrap();
+
+  /**
+   * The unqualified, case-sensitive semantic message operation name. The currently defined names
+   * are "send", "receive", "bulk-send" and "bulk-receive".
+   *
+   * <p>Examples:
+   * <pre><ul>
+   *   <li>Amazon SQS - {@code AmazonSQS.sendMessageBatch()} returns "bulk-send" as it can send many messages</li>
```

Comment:
ps I'm still on the fence about bulk-X vs X-bulk :P

Reply:
I'm thinking `send-bulk`